src/silicone/database_crunchers/quantile_rolling_windows.py

```diff
@@ -190,12 +196,20 @@ def derive_relationship(
             if max(xs) == min(xs):
                 # We must prevent singularity behaviour if all the points are at the
                 # same x value.
-                cumsum_weights = np.array([(1 + x) / len(ys) for x in range(len(ys))])
+                cumsum_weights = np.array([(0.5 + x) / len(ys) for x in range(len(ys))])
 
                 def same_x_val_workaround(
                     _, ys=ys, cumsum_weights=cumsum_weights, quantile=quantile
                 ):
-                    return min(ys[cumsum_weights >= quantile])
+                    if min(ys) == max(ys):
```

Comment:
```suggestion
                    if np.equal(min(ys), max(ys)):
```

safer when dealing with numeric types, `==` can evaluate to `True` in odd cases

Reply:
ok, I have assumed that this also applies about 8 lines earlier. I don't understand why that would ever happen though.